src/IRutils.cpp

```diff
@@ -365,36 +365,36 @@ std::string resultToSourceCode(const decode_results *results) {
   // Only display the value if the decode type doesn't have an A/C state.
   if (!hasACState(results->decode_type))
     output += " " + uint64ToString(results->value, 16);
-  output += "\n";
+  output += F("\n");
 
   // Now dump "known" codes
   if (results->decode_type != UNKNOWN) {
     if (hasACState(results->decode_type)) {
 #if DECODE_AC
       uint16_t nbytes = results->bits / 8;
-      output += "uint8_t state[" + uint64ToString(nbytes) + "] = {";
+      output += String(F("uint8_t state[")) + uint64ToString(nbytes) + String(F("] = {"));
```

Comment:
Why are you using `String()` here? this will cause the unit tests to break (`String()` is only available in the Arduino Framework.) Does it not compile under the Arduino IDE with out it?

E.g. https://travis-ci.org/markszabo/IRremoteESP8266/jobs/518287553#L7282

Reply:
`F("something")+ whatever ` does not compile. With `String(F("something")` it does. If you wish i will revert or refactor those lines
```
output += String(F("uint8_t state["));
output += uint64ToString(nbytes);
output += String(F("] = {"));
```

Reply:
And this is one of the problems with the use of the `F()` macro. :-/
Yeah, if it doesn't compile due to incompatible type issues we'll need to break it down to:

```
output += F("uint8_t state[");
output += uint64ToString(nbytes);
output += F("] = {");
```

How does that work/compile?

Reply:
yes that should work. I will do this in the coming days.

Reply:
No worries. Good luck. Take your time.